wiki/docs/developer-resources/functional-spec.md

```diff
@@ -0,0 +1,218 @@
+
+Here are some examples of settings per view. This is not exhaustive.
+
+##### Clock
+
+|Setting name|Required yes/no|Type|Default value|Notes|
+|---|---|---|---|---|
+|Clock type|Yes|Enum (big, small, animated)|big|
+|Show weather|Yes|Boolean|True|
+|Show date|Yes|Boolean|True|
+|Animation speed (s)|No|Whole number|30|Applies only if clock type is set to animated|
+
+![Clock settings](functional-spec-images/clock-settings.png)
+
+##### Weather
+
+|Setting name|Required yes/no|Type|Default value|Notes|
+|---|---|---|---|---|
+|Weather entity|Yes|HA entity name in `weather` domain|If only one entity in `weather` domain then that is the default, otherwise None|Weather entity to use to show current weather and forecast with. This should be a dropdown of HA entities in the `weather` domain.|
+|Show forecast|Yes|Boolean|True|
+
+##### Adding a view
+WHen a user selects the add view button in the tabstrip containing the settings pages, the add view page opens:
+![Add view](functional-spec-images/add-view.png)
+
+Here, the user can add a new view by entering a unique view name, it's view type (any of the built-in view types or custom) and the access sentences. If any of these sentences are uttered the view should open.
+If the user sets the view type to custom the view code box shows in which the user can type/paste YAML code for the view.
+Add and cancel buttons are provided. If the user selects 'add view' the view is added to the tabstrip and if it is a built-in view its configuration settings are shown on that page.
+
+##### Access control settings
+The access control settings page allows the user to determine which views are accessible to which satellites. By default all views are available to all satellites. By disabling the correct checkbox a view can be made inaccessible to a satellite, which means that that satellite cannot access the view even if the correct view access sentence is uttered. The default view is always accessible by all satellites.
+
+![Access control](functional-spec-images/access-control.png)
+
+### Panel behavior
+1. By default the `default view` is shown on the satellite's display.
+1. The user can activate other views by uttering access sentences. The view is shown on the satellite's device that the user is using.
+1. After `timeout` second the satellite in use switches back to the `default view`.
+1. Access sentences can be used while on any view, even while music is playing. If at all possible we would want to detect intents from the configured `assistant` and subscribe to handle those, so we do not need any automations.
+1. Something about link icons goes here, not sure yet what they are and how they fit in.
+
+## Development plan and priorities
+|Workitem #|Workitem|Priority|Pre requisites|Exit criteria|
+|---|---|---|---|---|
+|1|Integration and entity registration|0|None|User can select, load VA integration and add one or more satellites.|
```

Comment:
When ready, let's split the integration piece up into seperate tasks using GH projects. We can do the same for the following work items too.